fedimint-server/src/config.rs

```diff
@@ -64,52 +77,152 @@ pub struct Peer {
     pub federation_name: String,
     pub bitcoind_rpc: String,
+
+    /// extra options for extra settings and modules
+    pub other: BTreeMap<String, serde_json::Value>,
 }
 
-#[async_trait(?Send)]
-impl GenerateConfig for ServerConfig {
-    type Params = HashMap<PeerId, ServerConfigParams>;
-    type ClientConfig = ClientConfig;
-    type ConfigMessage = (KeyType, DkgMessage<G1Projective>);
-    type ConfigError = ();
+impl ServerConfig {
+    pub fn to_client_config(&self) -> ClientConfig {
+        let nodes = self
+            .peers
+            .iter()
+            .map(|(peer_id, peer)| Node {
+                url: peer.api_addr.clone(),
+                name: format!("node #{}", peer_id),
+            })
+            .collect();
+
+        ClientConfig {
+            federation_name: self.federation_name.clone(),
+            nodes,
+            modules: self
+                .modules
+                .iter()
+                .map(|(name, cfg)| {
+                    (
+                        name.clone(),
+                        match name.as_str() {
+                            "wallet" => cfg
+                                .cast::<WalletConfig>()
+                                .expect("valid config should not fail")
+                                .to_client_config(),
+                            "mint" => cfg
+                                .cast::<MintConfig>()
+                                .expect("valid config should not fail")
+                                .to_client_config(),
+                            "ln" => cfg
+                                .cast::<LightningModuleConfig>()
+                                .expect("valid config should not fail")
+                                .to_client_config(),
+                            other => panic!("module name that was not expected: {other}"),
+                        },
+                    )
+                })
+                .collect(),
+        }
+    }
+
+    pub fn validate_config(&self, identity: &PeerId) -> anyhow::Result<()> {
+        if self.epoch_sks.public_key_share()
+            != self.epoch_pk_set.public_key_share(identity.to_usize())
+        {
+            bail!("Epoch private key doesn't match pubkey share");
+        }
+        if self.hbbft_sks.public_key_share()
+            != self.hbbft_pk_set.public_key_share(identity.to_usize())
+        {
+            bail!("HBBFT private key doesn't match pubkey share");
+        }
+        if self.peers.keys().max().copied().map(|id| id.to_usize()) != Some(self.peers.len() - 1) {
+            bail!("Peer ids are not indexed from 0");
+        }
+        if self.peers.keys().min().copied() != Some(PeerId::from(0)) {
+            bail!("Peer ids are not indexed from 0");
+        }
 
-    fn trusted_dealer_gen(
+        let res: anyhow::Result<Vec<()>> = self
+            .modules
+            .iter()
+            .map(|(name, cfg)| match name.as_str() {
+                "wallet" => cfg
```

Comment:
Add a comment to do something about hardcodings after modularization is complete.